src/rrdp.rs

```diff
@@ -663,15 +655,31 @@ mod serde_current_object_map {
 
 //------------ CurrentObject -------------------------------------------------
 
-#[derive(Clone, Debug, Deserialize, Eq, Hash, Serialize, PartialEq)]
+#[derive(Clone, Debug, Deserialize, Eq, Serialize, PartialEq)]
 pub struct CurrentObject {
     uri: Rsync,
 
     #[serde(deserialize_with = "util::de_bytes", serialize_with = "util::ser_bytes")]
     data: Bytes,
+    since: Time,
 }
 
 impl CurrentObject {
+    pub fn new(
+        uri: Rsync,
+        data: Bytes,
+    ) -> Self {
+        let mut object =  CurrentObject {
+            uri, data, since: Time::now()
+        };
+
+        if let Err(e) = object.fix_since() {
+            warn!("Could not guess creation time for object at uri: {}. Error: {}", object.uri(), e);
```

Comment:
Guess? It's not trying to guess. Maybe this would be a place to use "derive"? ;-)

Reply:
Sure.. "derive" it is.. technically though, it is a guess because objects are quite often intentionally back-dated a few minutes. Doesn't really matter though.